### ISSUE TITLE
/account/password/msisdn/requestToken must die

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix-doc#2303]
using a msisdn to authenticate a password reset is a silly idea, because msisdn recycling is a thing

**Comment by turt2live** [comment]
I started writing a MSC for this and I'm not entirely convinced. The `/account/password` endpoint supports UIA so it would be in the server's best interest to require multiple stages. Not to mention our competitors support phone number password resets, so why shouldn't we? 

It seems completely reasonable for Synapse/Riot to refuse support though, imo.

**Comment by aaronraimist** [comment]
> Not to mention our competitors support phone number password resets, so why shouldn't we?

Seems like a great reason NOT to support that. Phone numbers are only somewhat useful as a second factor in two factor authentication. Having access to just a phone numbers should not allow you to reset a password, that is very insecure with all of these SIM swapping issues going around. Just like a week or two ago the Twitter CEO’s @jack account was taken over because Twitter allows you to control an account with just a phone number. The spec should not allow that.

**Comment by KitsuneRal** [comment]
As long as ownership of MSISDN is maintained, that's too useful to forfeit. Recycling email addresses is a thing as well; and to take over an email is arguably even easier than to take over a phone number. Unfortunately, we don't have any 3PID source as equally widespread as these two.

**Comment by richvdh** [comment]
yeah, having thought about this a bit more, I'm not sure that we need to actually kill this off from the spec, though I think it's relevant that synapse has never supported it.

ideally the spec would be entirely agnostic about the 3pid type - so it wouldn't matter from the spec's point of view whether the 3pid was `email`, `msisdn`, or `icq number`, and it would be entirely up to server implementations to figure out which 3pids it wants to accept for password resets.

**Comment by turt2live** [comment]
The spec is supposed to be 3PID agnostic, but for documentation reasons it breaks it out. This could probably be made clearer, somehow.
